src/ray/core_worker/core_worker.cc

```diff
@@ -1545,16 +1547,27 @@ std::vector<rpc::ObjectReference> CoreWorker::SubmitTask(
 
 Status CoreWorker::CreateActor(const RayFunction &function,
                                const std::vector<std::unique_ptr<TaskArg>> &args,
-                               const ActorCreationOptions &actor_creation_options,
+                               const ActorCreationOptions &options,
                                const std::string &extension_data,
                                ActorID *return_actor_id) {
+  auto actor_creation_options = options;
```

Comment:
I think is better not to copy the whole options. Instead, try something like this:

```cpp
bool is_detached;
if (!actor_creation_options.is_detached.has_value()) {
  /// Since this actor doesn't have a specified lifetime on creation, let's use
  /// the default value of the job.
  is_detached = job_config_->default_actor_lifetime() == ray::rpc::JobConfig_ActorLifetime_DETACHED;
} else {
  is_detached = actor_creation_options.is_detached.value();
}
```

Then pass the local variable `is_detached` to `builder.SetActorCreationTaskSpec(...)` and `actor_manager_->AddNewActorHandle(...)`.

Reply:
Good idea.